rfcs/0002-authentication.md

```diff
@@ -1,12 +1,12 @@
 ---
-RFC: 0002
+RFC: 2
```

Comment:
:+1: 

@kylef I am up for merging this but I would stick with the leading zeroes as @smizell suggests